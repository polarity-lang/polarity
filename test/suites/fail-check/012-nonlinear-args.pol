data Bool { T, F }

data Eq(a: Type, x y: a) {
    Refl(a: Type, x: a): Eq(a, x, x)
}

data Foo(a b: Bool) {}

def Eq(Bool, b1, b2).bar(b1 b2: Bool, x: Foo(?, b2)): Foo(b1, b2) {
    Refl(_, _) => x
}
