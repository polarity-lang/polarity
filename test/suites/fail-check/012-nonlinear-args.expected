T-020

  × The metavariable ?0 received b2 as an argument more than once
    ╭─[012-nonlinear-args.pol:9:46]
  8 │ 
  9 │ def Eq(Bool, b1, b2).bar(b1 b2: Bool, x: Foo(?, b2)): Foo(b1, b2) {
    ·                                              ─
 10 │     Refl(_, _) => x
    ·                   ┬
    ·                   ╰── While elaborating
 11 │ }
    ╰────
  help: This means that the metavariable cannot be solved automatically.
